src/codegeneration/InstantiateModuleTransformer.js

```diff
@@ -250,6 +253,18 @@ export class InstantiateModuleTransformer extends ModuleTransformer {
     this.exportStarBindings = [];
   }
 
+  transformIdentifierExpression(tree) {
+    if (tree.identifierToken.value == '__moduleName')
+      this.usesModuleName = true;
+    return super.transformIdentifierExpression(tree);
+  }
+
+  transformBindingIdentifier(tree) {
```

Comment:
This should not be overridden. BindingIdentifiers introduce new bindings... Like so:

```js
function x() {  // x: BindingIdentifier
  var y = 42;  // y: BindingIdentifier
  return z;  // z: IdentifierExpression
}
```

It is enough to override `transformIdentifierExpression` even though it will give a few false positives (which are harmless in this case).